lib/interfaces/common.js

```diff
@@ -109,7 +110,10 @@ module.exports = function (suites, context, mocha) {
           suite.parent._onlySuites = suite.parent._onlySuites.concat(suite);
         }
         if (typeof opts.fn === 'function') {
-          opts.fn.call(suite);
+          var result = opts.fn.call(suite);
+          if (result) {
+            utils.deprecate('Returning a value from a Suite is deprecated; this will throw an exception in the next major version of Mocha');
```

Comment:
As we know `async` functions will return a `<Promise>` however this might not be clear to the average user and thus the message will be vague (they are not returning anything yet this error is stating that they are).

> utils.deprecate('Returning a value from a Suite is deprecated; this will throw an exception in the next major version of Mocha');

Maybe we can reword to something in this context:

```
utils.deprecate('Deprecation Warning: Suites do not support a return value;' + suite.title + 'returned :' + result);
```

Reply:
Makes sense. Updated.